x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -427,26 +462,31 @@ private LogicalPlan resolveKeep(Project p, List<Attribute> childOutput) {
             }
             // otherwise resolve them
             else {
-                var starPosition = -1; // no star
-                // resolve each item manually while paying attention to:
-                // 1. name patterns a*, *b, a*b
-                // 2. star * - which can only appear once and signifies "everything else" - this will be added at the end
-                for (var ne : projections) {
-                    if (ne instanceof UnresolvedStar) {
-                        starPosition = resolvedProjections.size();
-                    } else if (ne instanceof UnresolvedAttribute ua) {
-                        resolvedProjections.addAll(resolveAgainstList(ua, childOutput));
-                    } else {
-                        // if this gets here it means it was already resolved
-                        resolvedProjections.add(ne);
+                Map<NamedExpression, Integer> priorities = new LinkedHashMap<>();
+                for (Attribute attribute : childOutput) {
```

Comment:
One concern that I have is the increased complexity of this solution versus to what we had (it is true that the previous one didn't have the capabilities this one has). I am wondering if we can do better that two nested loops.

Reply:
I'll try to see if we can do better. Anyway, this is not a critical path (it's a single execution per query, even if we have thousands of fields and a few expressions it will still take microseconds), so I tend to prefer a solution that is easy to read and understand rather than save a few machine cycles.

Reply:
I thought a bit about this, but I can't find a direction to try to optimize it.
The only idea I have is to loop through inputs and expressions separately, eg. to create and minimize disjoint groups based on wildcard expressions (Eg. `KEEP fo*, bar, baz, foo*` can be reduced to `KEEP bar, baz, foo*`), but there are cases where overlaps cannot be calculated (eg. `KEEP first*, *name`) so it's practically impossible to generalize.
The only other idea is to do it right to left (since last one wins), but it will need yet another check in the list to see if the column was added already, the computational complexity will be the same and probably it will be harder to understand, so my feeling is that we could leave it as it is for now and maybe try another round of optimizations at a later moment, when we have more time.